src/test/java/mate/academy/onlinebookstore/service/book/BookServiceImplTest.java

```diff
@@ -0,0 +1,165 @@
+package mate.academy.onlinebookstore.service.book;
+
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.onlinebookstore.dto.book.BookDto;
+import mate.academy.onlinebookstore.dto.book.CreateBookRequestDto;
+import mate.academy.onlinebookstore.mapper.BookMapper;
+import mate.academy.onlinebookstore.model.Book;
+import mate.academy.onlinebookstore.model.Category;
+import mate.academy.onlinebookstore.repository.book.BookRepository;
+import mate.academy.onlinebookstore.repository.category.CategoryRepository;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+public class BookServiceImplTest {
```

Comment:
the same. Try to simplify your test cases. Also, you can use Given-When-Then construction.